src/elm/Logs.elm

```diff
@@ -469,3 +493,53 @@ toString log =
 
         Nothing ->
             ""
+
+
+
+-- ANSI
+
+
+{-| defaultLogModel : struct to represent default model required by ANSI parser
+-}
+defaultLogModel : Ansi.Log.Model
+defaultLogModel =
+    { lineDiscipline = Ansi.Log.Cooked
+    , lines = Array.empty
+    , position = defaultPosition
+    , savedPosition = Nothing
+    , style = defaultLogStyle
+    , remainder = ""
+    }
+
+
+{-| defaultLogStyle : struct to represent default style required by ANSI model
+-}
+defaultLogStyle : Ansi.Log.Style
+defaultLogStyle =
+    { foreground = Nothing
+    , background = Nothing
+    , bold = False
+    , faint = False
+    , italic = False
+    , underline = False
+    , blink = False
+    , inverted = False
+    , fraktur = False
+    , framed = False
+    }
+
+
+{-| defaultPosition : default ANSI cursor position
+-}
+defaultPosition : Ansi.Log.CursorPosition
+defaultPosition =
+    { row = 0
+    , column = 0
+    }
+
+
+{-| decodeAnsi : takes maybe log parses into ansi decoded log line array
+-}
+decodeAnsi : Maybe (WebData Log) -> Array.Array Ansi.Log.Line
+decodeAnsi log =
+    .lines <| Ansi.Log.update (decodeLog log) defaultLogModel
```

Comment:
this is where the magic happens 🤘 
takes a default empty ansi log model, parses out the encoded style, and returns a list of lines with applied style actions, ready to be rendered

see: https://package.elm-lang.org/packages/vito/elm-ansi/latest/Ansi-Log#update